sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Tests/E2ETests/AspNetCoreInstrumentationTests.cs

```diff
@@ -0,0 +1,215 @@
+                            options.EnrichWithException = (activity, exception) => { activity.SetTag("enrichedOnException", "exception"); };
+                        });
+                    });
+
+                    builder.Configure(app =>
+                    {
+                        app.UseRouting();
+
+                        app.UseEndpoints(endpoints =>
+                        {
+                            endpoints.MapGet("/custom-endpoint", async context =>
+                            {
+                                context.Response.StatusCode = statusCode;
+                                await context.Response.WriteAsync("Hello!");
+                            });
+
+                            endpoints.MapGet("/exception-endpoint", context =>
+                            {
+                                context.Response.StatusCode = statusCode;
+                                throw new Exception("test exception");
+                            });
+                        });
+                    });
+                })
+                .CreateClient())
+            {
+                // Act
+                string url = queryString is null
+                    ? path
+                    : path + queryString;
+
+                expectedUrl = new Uri(client.BaseAddress!, url).AbsoluteUri;
+
+                try
+                {
+                    using var response = await client.GetAsync(url);
+                }
+                catch
+                {
+                    // Ignore exceptions
+                }
+
+                WaitForActivityExport(activities);
+                WaitForActivityExport(telemetryItems, x => x.Name == "Request");
+            }
+
+            // TODO: CHANGE METRIC COLLECTION FROM DEFAULT 1 MINUTE
+
+            // ASSERT
+            var activity = activities.Single();
+            Assert.True(telemetryItems.Any(), "Unit test failed to collect telemetry.");
+            var telemetryItem = telemetryItems.Where(x => x.Name == "Request").Single();
+
+            VerifyTelemetryItem(
+                isSuccess: statusCode == 200,
+                isException: shouldThrow,
+                expectedUrl: expectedUrl,
+                operationName: $"GET {path}",
+                statusCode: statusCode.ToString(),
+                telemetryItem: telemetryItem,
+                activity: activity);
+        }
+
+        private void WaitForActivityExport<T>(List<T> traceTelemetryItems, Func<T, bool>? predicate = null)
+        {
+            var result = SpinWait.SpinUntil(
+                condition: () =>
+                {
+                    Thread.Sleep(10);
+                    return predicate is null
+                        ? traceTelemetryItems.Any()
+                        : traceTelemetryItems.Any(predicate);
+                },
+                timeout: TimeSpan.FromSeconds(20));
```

Comment:
Is there a reason for long time-out?

Reply:
Half the tests failed on the CI because no activity was exported. 
I just increased the timeout in the last commit to see if they pass when given extra time.

Reply:
now that tests are passing, i've dropped this back down to 10 seconds.